### PR TITLE
Login with Token

### PR DESCRIPTION
**Pull request by UladBohdan**
This is to enable further Fibery integrations.

TODO:
- [x] fix a problem with token auth and session auth conflicting

### CODE REVIEW

**Review by Drapegnik — DISMISSED** [review]

File: src/utils/validation.js [hunk 1]
```diff
@@ -145,6 +145,18 @@ export const requireFields = (...fields) => (req, res, next) => {
...
+export const requireFieldsAny = (...fields) => (req, res, next) => {
+  const errors = {};
+
+  if (!Object.keys(req.body).some(field => fields.includes(field))) {
+    fields.forEach(field => {
+      errors[field] = 'errors.fieldListOneOfRequired';
+    });
+  }
+
+  return next(!isEmpty(errors) && new ValidationError(errors));
```
Comment (Drapegnik) [inline src/utils/validation.js:157]: `next(false)` - not interpreted as an error?

File: src/utils/validation.js [hunk 2]
```diff
@@ -145,6 +145,18 @@ export const requireFields = (...fields) => (req, res, next) => {
   return next(!isEmpty(errors) && new ValidationError(errors));
 };
 
+export const requireFieldsAny = (...fields) => (req, res, next) => {
+  const errors = {};
+
+  if (!Object.keys(req.body).some(field => fields.includes(field))) {
```
Comment (Drapegnik) [inline src/utils/validation.js:151]:
```js
const hasFields = fields.some(field => req.body[field]);
if (!hasFields) {}
```

**Review by Drapegnik — CHANGES_REQUESTED** [review]

File: src/utils/validation.js (hunk 1 above)
Comment (UladBohdan) [reply src/utils/validation.js:157]: I took this from `requireFields()` function so I guess we are ok with that. Also I've tried this API: works fine on valid requests.

File: src/utils/validation.js (hunk 2 above)
Comment (UladBohdan) [reply src/utils/validation.js]: Fixed

**Review by Drapegnik — APPROVED** [review]

**Pull request merged**